### ISSUE TITLE
no-restricted-imports doesn't restrict export * when importNames are configured

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#12737]
**Tell us about your environment**

* **ESLint Version:** v6.8.0
* **Node Version:** v12.14.0
* **npm Version:** v6.13.4

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    parserOptions: {
        ecmaVersion: 2015,
        sourceType: "module"
    },
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

[Online Demo Link](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXJlc3RyaWN0ZWQtaW1wb3J0czogW1wiZXJyb3JcIixcbiAge1xuICAgIG5hbWU6IFwibW9kMVwiICBcbiAgfSxcbiAge1xuICAgIG5hbWU6IFwibW9kMlwiLFxuICAgIGltcG9ydE5hbWVzOiBbXCJhXCJdXG4gIH1cbl0qL1xuXG5pbXBvcnQgKiBhcyBmb28gZnJvbSBcIm1vZDFcIjsgLy8gZXJyb3JcbmV4cG9ydCAqIGZyb20gXCJtb2QxXCI7IC8vIGVycm9yXG5cbmltcG9ydCB7IGEgfSBmcm9tIFwibW9kMlwiOyAvLyBlcnJvclxuZXhwb3J0IHsgYSB9IGZyb20gXCJtb2QyXCI7IC8vIGVycm9yXG5pbXBvcnQgKiBhcyBiYXIgZnJvbSBcIm1vZDJcIjsgLy8gZXJyb3JcbmV4cG9ydCAqIGZyb20gXCJtb2QyXCI7IC8vIG5vIGVycm9yXG5cbiIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6Niwic291cmNlVHlwZSI6Im1vZHVsZSIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==)

```js
/* eslint no-restricted-imports: ["error",
  {
    name: "mod1"  
  },
  {
    name: "mod2",
    importNames: ["a"]
  }
]*/

import * as foo from "mod1"; // error
export * from "mod1"; // error

import { a } from "mod2"; // error
export { a } from "mod2"; // error
import * as bar from "mod2"; // error
export * from "mod2"; // no error, is this a bug?
```

```bash
eslint index.js
```

**What did you expect to happen?**

Errors for the first 5 declarations.

I'm not sure what should be expected behavior for `export * from "mod2"`. That is the question of this issue.

**What actually happened? Please include the actual, raw output from ESLint.**

Errors for the first 5 declarations, as expected.

No error for `export * from "mod2"`.

```
  11:1  error  'mod1' import is restricted from being used                no-restricted-imports
  12:1  error  'mod1' import is restricted from being used                no-restricted-imports
  14:1  error  'mod2' import is restricted from being used                no-restricted-imports
  15:1  error  'mod2' import is restricted from being used                no-restricted-imports
  16:1  error  * import is invalid because 'a' from 'mod2' is restricted  no-restricted-imports
```

**Are you willing to submit a pull request to fix this bug?**

Yes.

**Comment by kaicataldo** [comment]
It makes sense to me to treat `import * as bar from "mod2";` and `export * from "mod2";` similarly and error on the latter as well.

**Comment by mdjermanovic** [comment]
As for the PR, seems best to finish and merge #12711 first and then do the PR for this fix on top of the changes from #12711.

**Comment by mdjermanovic** [comment]
I'm working on this.
